cmd/locateCG/locateCG.go

```diff
@@ -0,0 +1,186 @@
+
+	//if number of provided args is not 2
+	if len(flag.Args()) != expectedNumArgs {
+		flag.Usage()
+		log.Fatalf("Error: expecting %d arguments, but got %d\n", expectedNumArgs, len(flag.Args()))
+	}
+
+	s := Settings{
+		InFa:      flag.Arg(0),
+		ChromName: flag.Arg(1),
+	}
+
+	//run locate() function on provided args; will output a slice of structs
+	locateCG(s)
+}
+
+func locateCG(s Settings) {
+	var err error
+	var output []CGsite
+	var f1, f2, s1, s2 dna.Base
+	var refBases, altBases string
+
+	//read in fasta file
+	f := fasta.Read(s.InFa)
+
+	//specify sequence 1 and 2
+	firstSeq := f[0].Seq
+	secondSeq := f[1].Seq
+
+	//if no sequences found
+	if len(firstSeq) == 0 || len(secondSeq) == 0 {
+		log.Fatalf("Missing or empty sequences for seq1 or seq2.")
+	}
+	//add check for equal lengths
+	if len(firstSeq) != len(secondSeq) {
+		log.Fatalf("Seq1 and seq2 not equal in length.")
+	}
+	//pre-allocate size of output
+	//output = make([]CGsite, 0, 10000)
+
+	//initialize pair of matched reference/alignment positions
+	refStart := 0
+	alnStart := 0
+
+	fmt.Printf("Beginning loop\n")
+	//for length of first sequence
+	for i := 0; i < len(firstSeq)-1; i++ {
+		//store dna.Bases
+		f1, f2 = firstSeq[i], firstSeq[i+1]
+		s1, s2 = secondSeq[i], secondSeq[i+1]
+		if !(dna.DefineBase(f1)) || !(dna.DefineBase(f2)) || !(dna.DefineBase(s1)) || !(dna.DefineBase(s2)) {
+			continue
+		}
+		refBases = ""
+		altBases = ""
+		if f1 != s1 {
+			refBases += dna.BaseToString(f1)
+			altBases += dna.BaseToString(s1)
+		}
+		if f2 != s2 {
+			refBases += dna.BaseToString(f2)
+			altBases += dna.BaseToString(s2)
+		}
+
+		//if there is a CG in sequence 1
+		if f1 == dna.C && f2 == dna.G {
+			//but no CG in sequence 2
+			if !(s1 == dna.C && s2 == dna.G) {
+				//convert alignment position to reference position
+				startPos := fasta.AlnPosToRefPosCounter(f[0], i, refStart, alnStart)
+				endPos := startPos + 1
+				//add to "gained CpGs"
+				output = append(output, CGsite{s.ChromName, startPos, endPos, "gain", i, (i + 1), refBases, altBases})
+			}
+			//if there is no CG in sequence 1 but there is a CG in sequence 2
+		} else if s1 == dna.C && s2 == dna.G {
+			//convert alignment position to reference position
+			startPos := fasta.AlnPosToRefPosCounter(f[0], i, refStart, alnStart)
+			endPos := startPos + 1
+			//add to "lost CpGs"
+			output = append(output, CGsite{s.ChromName, startPos, endPos, "loss", i, (i + 1), refBases, altBases})
+		}
+
+		//if a CG gain or loss has already been recorded in "output"
+		if len(output) > 0 {
+			//reset the refStart to the last reference position
+			refStart = output[len(output)-1].StartPos
+			//reset the alnStart to the last corresponding alignment position
+			alnStart = output[len(output)-1].AlnStart
+			//this will now be the new ref/aln position pair for conversion
+		}
+	}
+	fmt.Printf("Finished loop\n")
+
+	//create output file name and initialize file, write header line
+	outfileName := s.ChromName + "CGs.txt"
```

Comment:
For a future PR, you can consider writing an option for the user to specify -outfileName on the command line!